numpyro/distributions/util.py

```diff
@@ -174,14 +174,24 @@ def _standard_gamma_grad(sample, alpha):
     return grads.reshape(alpha.shape)
 
 
+def _standard_gamma_batching_rule(batched_args, batch_dims):
+    x, y = batched_args
+    bx, by = batch_dims
+    size = next(t.shape[i] for t, i in zip(batched_args, batch_dims)
+                if i is not None)
+    x = batching.bdim_at_front(x, bx, size, force_broadcast=True)
+    y = batching.bdim_at_front(y, by, size, force_broadcast=True)
+    return _standard_gamma_p(x, y), 0
+
+
 @custom_transforms
 def _standard_gamma_p(key, alpha):
     return _standard_gamma_impl(key, alpha)
 
 
 ad.defjvp2(_standard_gamma_p.primitive, None,
            lambda tangent, sample, key, alpha, **kwargs: tangent * _standard_gamma_grad(sample, alpha))
-batching.defvectorized(_standard_gamma_p.primitive)
+batching.primitive_batchers[_standard_gamma_p.primitive] = _standard_gamma_batching_rule
```

Comment:
I am guessing this is also an issue with other samplers like `binomial`, `poisson` etc.? Are there JAX samplers which require custom batching implementations too?

Reply:
I have not tested through those samplers. But vmap works for all of our examples. So I think we don’t need to do so for non-custom-transform samplers.